src/sharding/ShardingManager.js

```diff
@@ -101,6 +109,54 @@ class ShardingManager extends EventEmitter {
     return shard;
   }
 
+  /**
+   * Kills all Shards without them respawning.
+   */
+  killAll() {
+    let chk = false;
+    if (this.respawn) {
+      this.respawn = false;
+      chk = true;
+    }
+    for (const shard of this.shards.values()) shard.kill();
+    if (chk) {
+      this.respawn = true;
+    }
+  }
+
+  /**
+   * Checks the Shard amount, then respawns the correct amount of Shards.
+   * @param {number} [delay=5500] How long to wait in between spawning each shard (in milliseconds)
+   * @param {boolean} [waitForReady=true] Whether to wait for a shard to become ready before continuing to another
+   * @returns {Promise<Collection<number, Shard>>}
+   */
+  async rebalanceShards(delay = 5500, waitForReady = true) {
+    // Obtain/verify that the totalShards is set to 'auto' for shards to spawn
+    if (!this.autoShards) throw new TypeError('CLIENT_INVALID_OPTION', 'Set totalShards', 'to auto.');
+    this.totalShards = await Util.fetchRecommendedShards(this.token);
+    let chk = false;
+    if (this.respawn) {
+      this.respawn = false;
+      chk = true;
+    }
+
+    // Kill all shards
+    await Promise.all(this.shards.map(shard => shard.kill()));
+
+    // Spawn the shards
+    for (let s = 1; s <= this.totalShards; s++) {
+      const promises = [];
+      const shard = this.createShard();
```

Comment:
Wouldn't this spawn even more shards?
I think you have to clear `this.shards` before creating new shards to avoid this.

For example, you have 3 shards and want (or have to) go to 4:
This would now kill all shards and then spawn shard 4, 5, 6, and 7.
1, 2, and 3 will stay dead.

Clearing this.shards and using regular spawn method will probably work.